### PR TITLE
Start documentation for Client Helpers

### LINKED ISSUE #823: [BUG] what does circuitbreaker configuration have to do with JS client?
### What is the bug?

Reading https://opensearch.org/docs/latest/clients/javascript/#circuit-breaker I don't quite understand how it relates to the JS client.

### What is the expected behavior?
Didn't expect to find this section here, or expected it to explain what it has to do with the JS client.

### PR DESCRIPTION
**Pull request by robdasilva**
### Description
Adds documentation for the Client Helpers `bulk` method including newly added behavior in https://github.com/opensearch-project/opensearch-js/pull/263. 

### Issues Resolved
Fixes #823 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Naarcha-AWS** [comment]
@robdasilva: Are there any other helpers we need documented?

**Comment by robdasilva** [comment]
@Naarcha-AWS Yes, as far as I know, there would be three more apart from the `bulk` helper:
* `search` helper: https://github.com/opensearch-project/opensearch-js/blob/main/lib/Helpers.js#L62
* `scrollSearch` helper: https://github.com/opensearch-project/opensearch-js/blob/main/lib/Helpers.js#L86
* `msearch` helper: https://github.com/opensearch-project/opensearch-js/blob/main/lib/Helpers.js#L186

**Comment by hdhalter** [comment]
@kolchfa-aws - Fanit, can you please review this PR? It's related to clients. Thanks.

File: _clients/javascript/helpers.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+const result = await client.helpers.bulk({
+  datasource: documents,
+  onDocument (doc) {
+    return {
+      index: { _index: 'example-index' }
+    }
+  }
+})
+
+console.log(result)
```
Comment (kolchfa-aws) [inline _clients/javascript/helpers.md:34]: Lines 35-42 represent the fields you will see in the log, correct? If so, I would separate those from the code snippet and explain them below.

File: _clients/javascript/helpers.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+```
+client.helpers.bulk({
+  datasource: arrayOfDocuments,
+  onDocument (doc) {
+    return {
+      delete: { _index: 'example-index', _id: doc.id }
+    }
+  }
+})
+```
```
Comment (kolchfa-aws) [inline _clients/javascript/helpers.md:168]: @robdasilva thank you for contributing this doc. Looks good! I would like to reformat and reword a little bit to parallel our other documentation. Do you mind if I take over this PR to do this or would you like me to comment so you can do it yourself?

Comment (robdasilva) [reply _clients/javascript/helpers.md:168]: Either way is fine for me. Happy to help where I can. But absolutely happy to hand this over to you as well. It’s probably quicker for you to change as desired :)

Comment (kolchfa-aws) [reply _clients/javascript/helpers.md:168]: OK thank you - I will make the changes.

File: _clients/javascript/getting-started.md [hunk 3]
```diff
@@ -1,10 +1,11 @@
...
 layout: default
-title: JavaScript client
-nav_order: 100
+title: Getting started
+parent: JavaScript client
+nav_order: 1
 ---
 
-# JavaScript client
+# Getting started
```
Comment (dblock) [inline _clients/javascript/getting-started.md:8]: Getting Started (capitalized)

File: _clients/javascript/getting-started.md [hunk 4]
```diff
@@ -142,18 +143,22 @@ search().catch(console.log);
 
 ## Circuit Breaker
 
-The `memoryCircuitBreaker` parameter in the [Cluster Settings API]({{site.url}}{{site.baseurl}}/opensearch/rest-api/cluster-settings/) gives you the ability to reject large query responses where the size of the response could crash OpenSearch Dashboards. To set the Circuit Breaker setting, use the `POST _cluster/settings` API operation on your active JS OpenSearch cluster.
+Helper function to check if memory circuit breaker enabled and the response payload is too large to fit into available heap memory.
```
Comment (dblock) [inline _clients/javascript/getting-started.md:146]: This doesn't read very well. Maybe make a higher level section called "Client Options" and a sub-section "Circuit Breaker" and then this line can just be removed?

File: _clients/javascript/helpers.md [hunk 5]
```diff
@@ -0,0 +1,168 @@
+---
+layout: default
+title: Helpers
+parent: JavaScript client
+nav_order: 2
+---
+
+# Helpers
+
+This section is currently incomplete. It does not yet cover all helper methods and is missing documentation of various helper options.
```
Comment (dblock) [inline _clients/javascript/helpers.md:10]: I would remove this and just open an issue to complete it.

File: _clients/javascript/helpers.md [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+{: .warning }
+
+## Bulk helper
```
Comment (dblock) [inline _clients/javascript/helpers.md:13]: Capitalize Consistently

File: _clients/javascript/helpers.md [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+
+Running bulk requests can be complex due to the shape of the API, this helper aims to provide a nicer developer experience around the Bulk API.
```
Comment (dblock) [inline _clients/javascript/helpers.md:15]: Maybe simpler to say _The `bulk` helper simplifies making complex bulk API requests._?

**Review by dblock — COMMENTED** [review]
Good progress. Made some nitpicky suggestions.

File: _clients/javascript/getting-started.md (hunk 3 above)
Comment (kolchfa-aws) [reply _clients/javascript/getting-started.md:8]: Thank you for reviewing, @dblock! Capitalization is per our [style guide](https://github.com/opensearch-project/documentation-website/blob/main/STYLE_GUIDE.md). I will make all other suggested changes.

Comment (kolchfa-aws) [reply _clients/javascript/getting-started.md:8]: I have a new PR for this issue: #1358. I will close this PR and make the changes in the new one.

**Comment by kolchfa-aws** [comment]
Closed in favor of the new PR #1358.

**Pull request closed without merging**